### PR TITLE
Fixes for tests among which CanLoadMultipleDocumentsConcurrently

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Changes

- Make CI more reliable by fixing the stability of `CanLoadMultipleDocumentsConcurrently`
- Get rid of `CloseDocumentAndWaitAsync` since it doesn't make sense to hover over a closed document, which is what it was doing.

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyLanguageServer/Workspace/ProjectManagerDatabase.cs [hunk 1]
```diff
@@ -87,7 +87,8 @@ public class ProjectManagerDatabase : IProjectDatabase, IDisposable {
     public async Task<IdeState?> GetResolvedDocumentAsync(TextDocumentIdentifier documentId) {
       var manager = await GetProjectManager(documentId, false);
       if (manager != null) {
-        return await manager.GetSnapshotAfterResolutionAsync()!;
+        var result = await manager.GetSnapshotAfterResolutionAsync()!;
```
Comment (alex-chew) [inline Source/DafnyLanguageServer/Workspace/ProjectManagerDatabase.cs:90]: nit/non-blocking: why is this needed?

**Review by alex-chew — APPROVED** [review]

File: Source/DafnyLanguageServer/Workspace/ProjectManagerDatabase.cs (hunk 1 above)
Comment (keyboardDrummer) [reply Source/DafnyLanguageServer/Workspace/ProjectManagerDatabase.cs:90]: Woops, accidentally checked this in. Will remove in a follow-up
